### PR TITLE
Viewing room list cards

### PR DESCRIPTION
**Pull request by pvinis**
Keep in mind:
- Using cards from palette.
- Removed an unused file still left from storybook.
- Added `RelayEnvironmentProvider` to be able to use relay-hooks.
- Now with pagination.
- iPad layout supported.

- [x] Pair with @mdole on making some tests.
- [x] Update to palette after the pending one.

Screenshot 2020-07-07 at 15 14 54
Screenshot 2020-07-07 at 15 31 34
Screenshot 2020-07-07 at 15 31 49

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/ViewingRoom/Components/__tests__/ViewingRoomSmallCard-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by mdole** [comment]
huh @pvinis I checked out the branch and tried opening a VR from the sim - still took me to the screen within the app and not the web view https://share.getcloudapp.com/YEupoEdE

no clue what's up there. maybe ask the MX team tomorrow?

**Comment by pvinis** [comment]
It was that I didn't have the lab option enabled on my sim 🤦. Interesting how everything else worked so far with that being off. Anyway, this should be ready to be merged now 👍.

File: src/lib/Scenes/ViewingRoom/Components/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  textColor: string
+  color: string
+  borderColor?: string
+  style: StyleProp<ViewStyle>
+}
+
+export const Tag: React.FC<TagProps> = ({ text, textColor, color, borderColor, style }) => {
+  return (
+    <View
+      style={[{ borderRadius: 2, overflow: "hidden", borderWidth: 1 }, style, { backgroundColor: color, borderColor }]}
```
Comment (mdole) [inline src/lib/Scenes/ViewingRoom/Components/Tag.tsx:16]: I usually think of Styled Components as being our go-to for React styling (though I'm not sure if we do more inline in Eigen) - should we use it here?

**Comment by pvinis** [comment]
I actually forgot about this pr 😅.

hm I didn't think of using styled. I guess we can.

For tests, since we are moving these to palette I think we could add some tests here later but I don't have a specific plan.

I'm ok with either merging and doing they changes in palette, or closing and doing everything directly in palette.

**Comment by mdole** [comment]
Ah yeah, and I forgot we're moving all of this to Palette haha. I'm going to reassign to you and let you make that call, both are totally reasonable to me and I'm not sure what your plan is for Palette so I don't want to slow you down! Let me know if I can help though

**Comment by pvinis** [comment]
Sorry for the many changes after upgrading this Pr from a draft to a regular one. This should be ready now.

**Comment by damassi** [comment]
@mdole - can this be merged?

**Comment by mdole** [comment]
@damassi ah sorry I keep getting distracted mid-review. I'll have a couple notes up in the next ~20min

File: src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx [hunk 2]
```diff
@@ -1,67 +1,140 @@
...
+export const ViewingRoomsListContainer: React.FC<ViewingRoomsListProps> = props => {
+  const [queryData, { isLoading, hasMore, loadMore }] = usePagination(fragmentSpec, props.query)
+  const viewingRooms = extractNodes(queryData.viewingRooms)
+
+  const _loadMore = () => {
+    if (!hasMore() || isLoading()) {
+      return
+    }
+    loadMore(connectionConfig, PAGE_SIZE)
+  }
```
Comment (mdole) [inline src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx:56]: @ds300 would love to get your 🧠  on this one - for some reason, after visiting the page, it keeps fetching more and more pages even when you're not near the bottom. Any idea what might be up?

File: tsconfig.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
     "strictFunctionTypes": true,
     "target": "es6",
     "types": ["jest", "react", "react-native"],
+    "skipLibCheck": true,
```
Comment (mdole) [inline tsconfig.json:23]: hm what's this do?

File: src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx (hunk 2 above)
Comment (ds300) [reply src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx:56]: hmm can't think of anything, happy to help debug it though.

File: tsconfig.json (hunk 3 above)
Comment (ds300) [reply tsconfig.json:23]: this prevents tsc from throwing errors found in `.d.ts` files. We probably should not have this switched on 👀 , curious why it was added here, @pvinis ?

File: src/lib/AppRegistry.tsx [hunk 4]
```diff
@@ -1,5 +1,7 @@
+import { defaultEnvironment } from "lib/relay/createEnvironment"
 import React from "react"
 import { AppRegistry, View, YellowBox } from "react-native"
+import { RelayEnvironmentProvider } from "relay-hooks"
```
[the remark this answers is not in the document]
Comment (pvinis) [reply src/lib/AppRegistry.tsx:4]: From looking around in the issues etc, I understand that `react-relay` doesn't have the hooks officially, and we can get them from `relay-hooks`. Apparently there is the `experimental` version of `react-relay` that has it. Let's add the dep for now, and if at some point they all exist in `react-relay`, we can remove it.

File: src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx (hunk 2 above)
Comment (pvinis) [reply src/lib/Scenes/ViewingRoom/ViewingRoomsList.tsx:56]: Ah it's fixed :D this part of the code was fine. The problem was caused from the FlatList reporting it's half-way down when it wasn't. All fixed by actually removing the parent ScrollView, so FlatList has better control and knowledge of the screen position and scroll etc.

File: tsconfig.json (hunk 3 above)
Comment (pvinis) [reply tsconfig.json:23]: Hm, I was getting errors coming from `node_modules`. Let me check again and report back.

Comment (pvinis) [reply tsconfig.json:23]: Screenshot 2020-07-13 at 12 00 54

Comment (pvinis) [reply tsconfig.json:23]: Hm I might use patch-package to fix their types?

Comment (pvinis) [reply tsconfig.json]: Ok I fixed the errors with a patch, and created an (issue)[https://github.com/relay-tools/relay-hooks/issues/99].

Comment (ds300) [reply tsconfig.json]: nice one!

**Comment by pvinis** [comment]
With everything solved, I'll go ahead and mark this #mergeongreen.

**Comment by ds300** [comment]
@pvinis for important new dependencies we normally talk about them in an RFC before adding them to eigen. I think relay-hooks definitely fits the bill. Do you mind submitting an RFC retroactively?

**Comment by pvinis** [comment]
Oh sure. I thought it's small enough, as it's basically some hooks for relay that we already have.

**Review by Brandylee24 — APPROVED** [review]

**Pull request merged**